CHANGELOG.md

```diff
@@ -6,6 +6,33 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [3.0.0]
+### Added
+- Add logging functions ([#20](https://github.com/MetaMask/utils/pull/20))
+- Add frozen collections (implemented in [#5](https://github.com/MetaMask/utils/pull/5) but exported in [#19](https://github.com/MetaMask/utils/pull/19))
+
+### Changed
+- **BREAKING:** Improve types and type validation ([#19](https://github.com/MetaMask/utils/pull/19))
+  - Various type changes have been made that might be breaking:
+    - The `JsonRpcRequest` and `JsonRpcNotification` types now include a generic constraint requiring that the `Params` type extends the `JsonRpcParams` type.
+    - The `JsonRpcSuccess` and `JsonRpcResponse` types now include a generic contraint for the `Result` type, requiring that it extends the `Json` type.
+    - Various validation functions now accept `unknown` parameters rather than specific types. This should not be breaking except that it may affect type inference for the parameters passed in.
+  - New JSON-related functions have been added:
+    - `assertIsJsonRpcResponse`
+    - `isJsonRpcResponse`
+    - `InferWithParams`
+    - `JsonRpcParams`
+  - New JSON Struct types have been added:
+    - `JsonRpcErrorStruct`
+    - `JsonRpcFailureStruct`
+    - `JsonRpcIdStruct`
+    - `JsonRpcParamsStruct`
+    - `JsonRpcRequestStruct`
+    - `JsonRpcResponseStruct`
+    - `JsonRpcSuccessStruct`
+    - `JsonRpcVersionStruct`
+    - `JsonStruct`
```

Comment:
These shouldn't be used outside of this package, so maybe we shouldn't put them in the changelog either. All of them can be used through `isJson[Something]` and `assertIsJson[Something]`.

Reply:
Actually, now that I think about it, there may be some use cases for using them outside the package, i.e., extending a certain type.

Reply:
Yeah, I was unsure about that too but they are exported.

Reply:
I'm going to leave it in for now, but we can mark it as deprecated or remove it later if we feel the need